### ISSUE TITLE
Add service definitions and Bundle

### ISSUE THREAD

**Issue by greg0ire** [sonata-project/exporter#118]
A `Bundle` directory should be created under `src`, containing optional integration with Symfony. This does not make this library a bundle, it makes it contain a bundle. Everything symfony related should go in this directory.

Check list : 

- [x] create an exporter class (deprecating the one in core bundle)
- [x] create service definitions for the writers
- [x] create a bundle to load and configure the service definitions
- [x] create a compiler pass to inject new writers in the exporter

**Comment by Soullivaneuh** [comment]
> A Bundle directory should be created under src

A `src/Bridge/Symfony` folder instead. If someone want to add a silex or a laravel integration for example, this directory structure makes more sense.

**Comment by greg0ire** [comment]
It's more explicit, yeah, good Idea. There should still be a `Bundle` directory under it, because symfony service definitions may be used outside a symfony application (in an application that uses symfony/dependency-injection but not the rest, for instance)

**Comment by Soullivaneuh** [comment]
> There should still be a Bundle directory under it

Yeah, for the Bundle class. For the rest, it's depends AFAIK.

Ref: https://github.com/nexylan/paybox-direct/tree/master/src/Bridge/Symfony

**Comment by Soullivaneuh** [comment]
BTW, the `Exporter` class would be common to Silex and Symfony (if we provide Silex support).

**Comment by core23** [comment]
Why shouldn't mess this repository with everything someone COULD use. This would add many unused classes to the autoloader.

This is a library that contains library stuff, nothing else. We should create a separate bundle instead.

**Comment by greg0ire** [comment]
> This is a library that contains library stuff, nothing else. We should create a separate bundle instead. 

This is indeed cleaner, and can be done later if we really feel the library is getting crammed with lots of different integrations (which, let's be honest, probably won't be the case).

**Comment by Soullivaneuh** [comment]
> This is a library that contains library stuff, nothing else. We should create a separate bundle instead.

This depends of the integration size.

If the integration provide tons of services, commands or whatever, yes, we should move it on a separated bundle.

If this is just for provide one class as a service, this would be more painful to maintain than the "benefit" we will have with a separated project.

**Comment by greg0ire** [comment]
Reminder for me to follow up on https://github.com/sonata-project/SonataCoreBundle/pull/270

**Comment by greg0ire** [comment]
I'm starting to work on this on 1.x, because introducing new services is not a BC-break, but all this will have to be renamed according to the new namespace when merging in master :confused:

**Comment by rande** [comment]
I don't see the need for a bundle for those library. What kind of services do you plan to add ?

**Comment by greg0ire** [comment]
I plan to move the exporter service from SonataCoreBundle to here. Also, I plan to add services for the default writers, to make them configurable (just like [in this PR](https://github.com/sonata-project/SonataCoreBundle/pull/270))

**Comment by greg0ire** [comment]
See [the documentation I wrote](https://github.com/greg0ire/SonataCoreBundle/blob/44806c8d0695332cfbb593f061cdef5c97186a55/Resources/doc/reference/exporter.rst) for more details.

TL;DR you will:
- be able to configure options default writers
- be able to pick and choose default writers
- be able to add new writers to the list

**Comment by greg0ire** [comment]
@rande see linked PR. It is fully documented, so it should be very easy to understand.
